guides/source/caching_with_rails.md

```diff
@@ -270,7 +270,7 @@ simply be explicit in a comment, like:
 
 Sometimes you need to cache a particular value or query result instead of caching view fragments. Rails' caching mechanism works great for storing __any__ kind of information.
 
-The most efficient way to implement low-level caching is using the `Rails.cache.fetch` method. This method does both reading and writing to the cache. When passed only a single argument, the key is fetched and value from the cache is returned. If a block is passed, the result of the block will be cached to the given key and the result is returned.
+The most efficient way to implement low-level caching is using the `Rails.cache.fetch` method. This method does both reading and writing to the cache. When passed only a single argument, the key is fetched and value from the cache is returned. If a block is passed for given key which is not used for cache yet, then result of the block will be cached to that key and the result is returned otherwise value from the cache for that key is returned.
 
```

Comment:
@prathamesh-sonpatki Some new developers can not digest `cache miss` & `cache hit` terms easily on the first impression. Need to mention `cache key` explicitly I think. All other parts look good to me 👍 

How about this one (changed little bit)

```
If a block is passed, that block will be executed when given cache key is not found. The return value of the block will be written to the cache under the given cache key, and that return value will be returned. If cache key is found, then the cached value will be returned without executing the block.
```